### ISSUE TITLE
Should `FilterIterator::key()` and `FilterIterator::current()` be updated?

### ISSUE THREAD

**Issue by drupol** [vimeo/psalm#8149]
Hi,

Let's consider this example: https://psalm.dev/r/0011a796d7

Why PSalm is complaining about the return type of the methods `key()` and `current()` ?

When using a `FilterIterator`, `key()` is always returning a `TKey` and `current()` a `T`, no ?

**Comment by weirdan** [comment]
Method signature here must provide a **native** return type (`mixed` in this case). Otherwise you'd get `Deprecated: Return type of A::key() should either be compatible with IteratorIterator::key(): mixed, or the #[\ReturnTypeWillChange] attribute should be used to temporarily suppress the notice in /in/aPp49 on line 15`: https://3v4l.org/aPp49

**Comment by drupol** [comment]
You're right, they are actually correctly inferred: https://psalm.dev/r/bd3aad2689

Thanks for the clarification.

Closing now.
